src/managers/OnOfframpManager.sol

```diff
@@ -0,0 +1,134 @@
+        spoke = spoke_;
+        balanceSheet = balanceSheet_;
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Owner actions
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IUpdateContract
+    function update(PoolId poolId_, ShareClassId, /* scId */ bytes calldata payload) external {
+        require(poolId == poolId_, InvalidPoolId());
+        require(msg.sender == spoke, NotAuthorized());
+
+        uint8 kind = uint8(UpdateContractMessageLib.updateContractType(payload));
+
+        if (kind == uint8(UpdateContractType.UpdateAddress)) {
+            UpdateContractMessageLib.UpdateContractUpdateAddress memory m =
+                UpdateContractMessageLib.deserializeUpdateContractUpdateAddress(payload);
+            address who = m.who.toAddress();
+
+            if (m.kind == "onramp") {
+                onramp[who] = m.isEnabled;
+                emit UpdateOnramp(who, m.isEnabled);
+            } else if (m.kind == "offramp") {
+                address asset = m.what.toAddress();
+                offramp[asset][who] = m.isEnabled;
+                emit UpdateOfframp(asset, who, m.isEnabled);
+            }
+        } else if (kind == uint8(UpdateContractType.Toggle)) {
+            UpdateContractMessageLib.UpdateContractToggle memory m =
+                UpdateContractMessageLib.deserializeUpdateContractToggle(payload);
+
+            if (m.what == "permissionlessOnramp") permissionlessOnramp = m.isEnabled;
+            else if (m.what == "permissionlessOfframp") permissionlessOfframp = m.isEnabled;
+            emit UpdatePermissionless(m.what, m.isEnabled);
+        } else {
+            revert UnknownUpdateContractType();
+        }
+    }
+
+    //----------------------------------------------------------------------------------------------
+    // Deposit & withdraw actions
+    //----------------------------------------------------------------------------------------------
+
+    /// @inheritdoc IDepositManager
+    function deposit(address asset, uint256, /* tokenId */ uint128 amount, address owner) external {
+        require(manager[msg.sender] || permissionlessOnramp, IAuth.NotAuthorized());
+
+        require(onramp[asset], NotAllowedOnrampAsset());
+        require(owner == address(this) || owner == msg.sender);
+        require(amount <= IERC20(asset).balanceOf(owner), InvalidAmount());
+
+        if (owner == msg.sender) SafeTransferLib.safeTransferFrom(asset, owner, address(this), amount);
+
+        balanceSheet.deposit(poolId, scId, asset, 0, amount);
+    }
+
+    /// @inheritdoc IWithdrawManager
+    function withdraw(address asset, uint256, /* tokenId */ uint128 amount, address receiver) external {
+        require(manager[msg.sender] || permissionlessOfframp, IAuth.NotAuthorized());
+        require(offramp[asset][receiver], InvalidOfframpDestination());
```

Comment:
```suggestion
        require(manager[msg.sender] || offramp[asset][receiver], IAuth.NotAuthorized());
```
Is it my suggestion equivalent to what exists without requiring explicitly to enable `permissionlessOfframp`?

Reply:
No if it's the manager, it should still be limted to the offramp destinations

The idea is that the manager can still be a single cold wallet, and doens't need to be a multisig. If we don't restrict the offramp destinations, it would need to be heavily restricted, and then we might as well make it a manager on the balance sheet.